vowpalwabbit/array_parameters.h

```diff
@@ -98,7 +99,10 @@ class sparse_parameters
   sparse_parameters(const sparse_parameters& other) { shallow_copy(other); }
   sparse_parameters(sparse_parameters&&) = delete;
 
-  weight* first() { throw 1; }  // TODO: Throw better exceptions. Allreduce currently not supported in sparse.
+  weight* first()
+  {
+    THROW_OR_RETURN_VOID("Allreduce currently not supported in sparse");
```

Comment:
This seems to be a weird macro name given that the function returns `weight*`. Is it the right one?

Reply:
Oh yes good catch, thanks. This is wrong